File: Greater on right side - GFG/greater-on-right-side.cpp
//{ Driver Code Starts

#include<bits/stdc++.h>

using namespace std;


// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	/* Function to replace every element with the
	next greatest element */
	void nextGreatest(int arr[], int n) {
	    int maxi=INT_MIN;
	    for(int i=n-1;i>=0;i--){
	        if(maxi==INT_MIN){
	            maxi=max(maxi,arr[i]);
	            arr[i]=-1;
	        }
	        else{
	            int v=arr[i];
	            arr[i]=maxi;
	            
	            maxi=max(maxi,v);
	            
	        }
	    }
	}


};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, l, r;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        ob.nextGreatest(arr, n);
        for (int i = 0; i < n; i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: Greater on right side - GFG/greater-on-right-side.java
//{ Driver Code Starts
//Initial Template for Java



import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine().trim());
        while (tc-- > 0) {
            String[] inputLine;
            int n = Integer.parseInt(br.readLine().trim());
            int[] arr = new int[n];
            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(inputLine[i]);
            }

            new Solution().nextGreatest(arr, n);
            for (int i = 0; i < n; i++) {
                System.out.print(arr[i] + " ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends


//User function Template for Java




class Solution {
    void nextGreatest(int arr[], int n) {
        int right=-1;
        for(int i=n-1;i>=0;i--){
            if(right==-1){
                right=arr[i];
                arr[i]=-1;
            }
            else{
               int  org_value=arr[i];
                arr[i]=right;
                
                right=Math.max(org_value,right);
            }
        }
    }
}